pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -364,7 +367,7 @@ func (lb *ListenerBuilder) buildHTTPConnectionManager(httpOpts *httpListenerOpts
 
 	// Metadata exchange filter needs to be added before any other HTTP filters are added. This is done to
 	// ensure that mx filter comes before HTTP RBAC filter. This is related to https://github.com/istio/istio/issues/41066
-	if features.MetadataExchange {
+	if features.MetadataExchange && !httpOpts.hbone {
```

Comment:
Are hbone and auto mtls exclusive? Here it looks so, however during cluster transportSocketmatch building, we can set 3 matches? So I am a little confused.

Reply:
I don't think they are exclusive - but the MetadataExchange is not part of hbone ( which uses Baggage header and hopefully a much lighter set of values ). The old MetadataExchange ( proto.Any in a header with a lot of info ) remains for the auto-mtls and sidecar-based istio.